app/components/video-block.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { createUseStyles } from 'react-jss'
+import ActionButton from './action-button'
+
+
+
+const VideoBlock = ({
+                      className,
+                      mode='dark',
+                      subject,
+                      actionText,
+                      action,
+                      videoUrl
+}) => {
+  const classes = useStyles({ mode })
+
+  const isYouTubeUrl = (url) => url.includes('youtube.com') || url.includes('youtu.be')
+
+  const getYouTubeEmbedUrl = (url) => {
+    const videoId = url.split('v=')[1] || url.split('/').pop()
+    return `https://www.youtube.com/embed/${videoId}`
+  }
+
+  return (
+    <div className={`${classes.videoBlock} ${className}`}>
+      {subject && <div className={classes.subject}>{subject}</div>}
+      <div className={classes.videoContainer}>
+        {isYouTubeUrl(videoUrl) ? (
+          <iframe
+            className={classes.iframe}
+            src={getYouTubeEmbedUrl(videoUrl)}
+            frameBorder="0"
+            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+            allowFullScreen
+            title="Embedded youtube"
+          ></iframe>
+        ) : (
+          <video className={classes.video} controls>
+            <source src={videoUrl} type="video/mp4" />
+            Your browser does not support the video tag.
+          </video>
+        )}
+      </div>
+      {actionText && action && (
+        <div className={classes.action}>
+          <ActionButton action={action}>
+            {actionText}
+          </ActionButton>
+        </div>
+      )}
+    </div>
+  )
+}
+
+VideoBlock.propTypes = {
+  className: PropTypes.string,
+  mode: PropTypes.oneOf(['dark', 'light']),
+  subject: PropTypes.string,
+  actionText: PropTypes.string,
+  action: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  videoUrl: PropTypes.string.isRequired,
+}
+
+VideoBlock.defaultProps = {
+  className: '',
+  mode: '',
+  subject: '',
+  actionText: '',
+  action: '',
+}
+
+export default VideoBlock
+
+
+const useStyles = createUseStyles(theme => ({
+  videoBlock: {
+    textAlign: 'center',
```

Comment:
From the figma, please figure out the padding to add to the top and bottom of this component, and around the Action Button, to make it look like what's shown in figma.

As for the width, please look at text-block.js and see the wrapper style, and how it's using theme.maxPanelWidth.